kythe/typescript/testdata/generic.ts

```diff
@@ -33,6 +33,27 @@ function generic<T>(x: T, y: IFace) {
   return x;
 }
 
+// Simple constrained generic.
+//- @T defines/binding ConstrainedGenericT
+//- @IFace ref IFace
+//- ConstrainedGenericT bounded/upper IFace
+function constrainedGeneric<T extends IFace>() {}
+
+// Constrained generic with reference to type param within it.
+//- @#0T defines/binding ConstrainedGenericT2
+//- !{@#0T ref ConstrainedGenericT}
+//- @#1T ref ConstrainedGenericT2
+function constrainedGenericRef<T, K extends keyof T>() {}
+
+// Recursive generic type.
+//- @#0T defines/binding ConstrainedGenericT3
+//- @#1T ref ConstrainedGenericT3
+function constrainedGenericRecursive<T extends Array<T>>() {}
+
+// Default generic.
+//- @IFace ref IFace
+function defaultGeneric<T = IFace>() {}
```

Comment:
What is the relationship between `T` and `IFace`? Can you add a verifier goal for the binding of `T`?

Reply:
IFace is the default value (er, type) of T.  I added a bit but I'm really sure what to assert.